sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala

```diff
@@ -162,12 +174,14 @@ case class BroadcastNestedLoopJoinExec(
             streamRow = streamedIter.next()
             nextIndex = 0
             foundMatch = false
+            matches = 0
           }
           while (nextIndex < buildRows.length) {
             resultRow = joinedRow(streamRow, buildRows(nextIndex))
             nextIndex += 1
             if (boundCondition(resultRow)) {
               foundMatch = true
+              matches = checkMatches(matches)
```

Comment:
Seems we only need a single `singleJoin` boolean flag?
```
if (boundCondition(resultRow)) {
  if (foundMatch && singleJoin) throw ...
  foundMatch = true
}
```

Reply:
neat!